### ISSUE TITLE
Tridiagonal/SymTridiagonal/Bidiagonal matrices don't convert to sparse matrices

### ISSUE THREAD

**Issue by bnels** [JuliaLang/julia#10837]
### Problem
It seems like it would be appropriate to be able to convert Tridiagonal/SymTridiagonal/Bidiagonal matrices to sparse matrices.  Currently:
 ```Julia
B = SymTridiagonal(randn(5),randn(4))
C = sparse(B)
```
Gives the following error:
```
LoadError: indexing not defined for Base.LinAlg.SymTridiagonal{Float64}
while loading In[20], in expression starting on line 2

 in count at reduce.jl:374
 in findnz at array.jl:1166
 in convert at sparse/sparsematrix.jl:167
 in sparse at sparse/sparsematrix.jl:234
```
Similar errors appear for
```Julia
B = Tridiagonal(randn(4),randn(5),randn(4))
C = sparse(B)
```
```Julia
B = Bidiagonal(randn(5),randn(4),true)
C = sparse(B)
```

A current workaround is to convert to a full matrix and then to a sparse matrix, but is slow and possibly infeasible for large matrices.

### Version Information
```Julia
versioninfo()
```
```
Julia Version 0.4.0-dev+3166
Commit e55de4b (2015-02-06 20:21 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i5-3230M CPU @ 2.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

### Context
I was trying to solve a Poisson equation on a 2D Finite Difference grid:
```Julia
M = 40
h = 1/M
I = speye(M-1)
d = 2*ones(M-1)
du = -1*ones(M-2)
B = SymTridiagonal(d, du)
A = kron(B,I) + kron(I,B)
A = 1/h^2 * A
```
`A` is the discretized Laplacian on a `MxM` grid on the unit square.
`kron` also does not work with these matrices, so I tried converting directly to sparse.

**Comment by andreasnoack** [comment]
We are still missing some conversions between all the special matrix types. Contributing them would be very welcome.

If you want a solution right now, I think the best to do is to construct `B` from the triplets, e.g. something like
```
B = sparse([1:M;2:M;1:M-1], [1:M;1:M-1;2:M], [2ones(M); -ones(2*(M-1))])
```

**Comment by bnels** [comment]
fixed in merge commit  2a3a217

**Comment by ViralBShah** [comment]
@bnels Thanks for the fix and welcome!
